ui/src/app/views/workspace/component.spec.ts

```diff
@@ -0,0 +1,109 @@
+import {Component, DebugElement} from '@angular/core';
+import {TestBed, async, fakeAsync, ComponentFixture} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+import {updateAndTick, simulateInput} from 'testing/test-helpers';
+import {CohortsServiceStub} from 'testing/stubs/cohort-service-stub';
+import {CohortsService} from 'generated';
+import {UserService} from 'app/services/user.service';
+import {RepositoryService} from 'app/services/repository.service';
+
+class Context {
```

Comment:
I do think we want the Context pattern because it lets us have a nice abstraction on what information is the test's and what information is the page's/component's. That is something I am happy to change if you feel strongly about though.